jbpm-system-test/jbpm-unmarshall-ejb-timers-test/src/test/java/org/kie/samples/integration/UnmarshallEJBTimerSystemTest.java

```diff
@@ -0,0 +1,220 @@
+        createCLIFile("node1");
+        createCLIFile("node2");
+    }
+
+    @ClassRule
+    public static Network network = Network.newNetwork();
+    
+    @ClassRule
+    public static PostgreSQLContainer<?> postgreSQLContainer = new PostgreSQLContainer<>(System.getProperty("org.kie.samples.image.postgresql","postgres:latest"))
+                                        .withDatabaseName("rhpamdatabase")
+                                        .withUsername("rhpamuser")
+                                        .withPassword("rhpampassword")
+                                        .withFileSystemBind("target/postgresql", "/docker-entrypoint-initdb.d",
+                                                            BindMode.READ_ONLY)
+                                        .withNetwork(network)
+                                        .withNetworkAliases("postgresql11");
+
+    @ClassRule
+    public static KieServerContainer kieServer1 = new KieServerContainer("node1", network, args);
+    
+    @ClassRule
+    public static KieServerContainer kieServer2 = new KieServerContainer("node2", network, args);
+    
+    private static KieServicesClient ksClient1;
+    private static KieServicesClient ksClient2;
+    
+    private static ProcessServicesClient processClient1;
+    private static ProcessServicesClient processClient2;
+    
+    private static HikariDataSource ds;
+    
+    @BeforeClass
+    public static void setup() {
+        logger.info("KIE SERVER 1 started at port "+kieServer1.getKiePort());
+        logger.info("KIE SERVER 2 started at port "+kieServer2.getKiePort());
+        logger.info("postgresql started at "+postgreSQLContainer.getJdbcUrl());
+        
+        ksClient1 = authenticate(kieServer1.getKiePort(), DEFAULT_USER, DEFAULT_PASSWORD);
+        ksClient2 = authenticate(kieServer2.getKiePort(), DEFAULT_USER, DEFAULT_PASSWORD);
+        
+        processClient1 = ksClient1.getServicesClient(ProcessServicesClient.class);
+        processClient2 = ksClient2.getServicesClient(ProcessServicesClient.class);
+        
+        createContainer(ksClient1);
+        createContainer(ksClient2);
+        
+        ds = getDataSource();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        ksClient1.disposeContainer(containerId);
+        ksClient2.disposeContainer(containerId);
+        DockerClient docker = DockerClientFactory.instance().client();
+        docker.listImagesCmd().withLabelFilter("autodelete=true").exec().stream()
+         .filter(c -> c.getId() != null)
+         .forEach(c -> docker.removeImageCmd(c.getId()).withForce(true).exec());
+    }
+
+    @Test
+    @DisplayName("user starts process in one node but get variable in another before refresh-time")
+    public void getVariableBeforeRefresh() throws Exception {
+        Long processInstanceId = startProcessInNode1("ht-timer-process");
+        
+        Thread.sleep(2000);
```

Comment:
not sure if we really need to sleep for 2 seconds here? duration variable in the process is set to 30secs, and am not sure about the refresh-time?

Reply:
not really needed (comes from the reproducer)